game/ui/city/weeklyfundingscreen.cpp

```diff
@@ -31,25 +38,25 @@ WeeklyFundingScreen::~WeeklyFundingScreen() = default;
 void WeeklyFundingScreen::begin()
 {
 	// Validate that we can recieve funding
-	if (state->fundingTerminated)
+	if (state.fundingTerminated)
 	{
 		fw().stageQueueCommand({StageCmd::Command::POP});
 		return;
 	}
 
-	menuform->findControlTyped<Label>("TEXT_FUNDS")->setText(state->getPlayerBalance());
-	menuform->findControlTyped<Label>("TEXT_DATE")->setText(state->gameTime.getLongDateString());
-	menuform->findControlTyped<Label>("TEXT_WEEK")->setText(state->gameTime.getWeekString());
+	menuform->findControlTyped<Label>("TEXT_FUNDS")->setText(state.getPlayerBalance());
+	menuform->findControlTyped<Label>("TEXT_DATE")->setText(state.gameTime.getLongDateString());
+	menuform->findControlTyped<Label>("TEXT_WEEK")->setText(state.gameTime.getWeekString());
 
 	menuform->findControlTyped<Label>("TITLE")->setText(tr("WEEKLY FUNDING ASSESSMENT"));
 
 	UString ratingDescription;
+	const Colour deepBlueColor = {4, 73, 130, 255};
+	const Colour cyanColor = {113, 219, 255, 255};
 
-	const auto player = state->getPlayer();
-	const auto government = state->getGovernment();
-	int currentIncome = player->income;
+	int currentIncome = state.playerIncome;
```

Comment:
Can be const?

Reply:
This one could change during the execution of the code (e.g: zeroed when funding are cut due to hostile relationships)